genomic_data_service/region_indexer.py

```diff
@@ -68,68 +73,224 @@
     }
 }
 
+FILE_REQUIRED_FIELDS = [
+    '@id',
+    '@type',
+    'uuid',
+    'assembly',
+    'accession',
+    'dataset',
+    'title',
+    'target',
+    'targets',
+    'href',
+    's3_uri',
+    'status',
+    'file_format',
+    'file_type',
+    'file_size',
+    'output_type',
+    'assay_term_name',
+    'annotation_type',
+    'reference_type'
+]
 
-ENCODE_QUERY = [
-    'internal_tags=RegulomeDB',
-    'field=files',
-    'field=biosample_ontology',
-    'field=target.name',
-    'field=biosample_term_name',
-    'field=documents',
-    'field=uuid',
-    'field=assay_term_name',
-    'field=annotation_type',
-    'field=reference_type',
-    'field=collection_type',
-    '&'.join([f'status={st}' for st in REGULOME_ALLOWED_STATUSES]),
-    'format=json',
-    'limit=all'
+DATASET_REQUIRED_FIELDS = [
+    '@id',
+    '@type',
+    'uuid',
+    'accession',
+    'target',
+    'targets',
+    'biosample_ontology',
+    'assay_term_name',
+    'annotation_type',
+    'reference_type',
+    'biosample_ontology',
+    'documents',
+    'status'
 ]
 
+def clean_up(obj, fields):
+    clean_obj = {}
+    keys = obj.keys()
+    for key in keys:
+        if key in fields:
+            clean_obj[key] = obj[key]
+    return clean_obj
+
+
+# https://stackoverflow.com/questions/3173320/text-progress-bar-in-the-console
+def print_progress_bar (iteration, total, prefix = 'Progress:', suffix = 'Complete', decimals = 1, length = 80, fill = '█', printEnd = "\r"):
+    percent = ("{0:." + str(decimals) + "f}").format(100 * (iteration / float(total)))
+    filledLength = int(length * iteration // total)
+    bar = fill * filledLength + '-' * (length - filledLength)
+
+    print(f'\r{prefix} |{bar}| {percent}% {suffix}', end = printEnd)
+
+    if iteration == total:
+        print()
+
 
 def encode_graph(query):
-    endpoint = f"https://www.encodeproject.org/search/?{'&'.join(query)}"
+    query += [
+        'field=*',
```

Comment:
Why do you list the required fields then search fields=*?

Reply:
`field=*` just makes it return the full JSON object, it does not act as a search param